frontend/src/components/forms/EditResident.tsx

```diff
@@ -0,0 +1,314 @@
+  const [moveInDateError, setMoveInDateError] = useState(false);
+  const [buildingError, setBuildingError] = useState(false);
+  const [moveOutDateError, setMoveOutDateError] = useState(false);
+  const [flagged, setFlagged] = useState(false);
+  const [alertData, setAlertData] = useState<AlertData>(ALERT_DATA.DEFAULT);
+
+  const [showAlert, setShowAlert] = useState(false);
+
+  const editRes = async () => {
+    await ResidentAPIClient.editResident({
+      id,
+      initial: initials.toUpperCase(),
+      roomNum: roomNumber,
+      dateJoined: moveInDate.toLocaleDateString('en-CA'),
+      building: userBuilding,
+      dateLeft: moveOutDate? moveOutDate.toLocaleDateString('en-CA') : undefined,
+    }).then((res) => {
+      if (res != null) {
+        setAlertData(ALERT_DATA.SUCCESS);
+      } else {
+        setAlertData(ALERT_DATA.ERROR);
+      }
+    });
+  };
+
+  const handleClear = () => {
+    setMoveOutDate(undefined);
+  };
+  
+  const handleInitialsChange = (e: { target: { value: unknown } }) => {
+    const inputValue = e.target.value as string;
+    if (/^[a-z]{0,2}$/i.test(inputValue)) {
+      setInitials(inputValue.toUpperCase());
+      setInitialsError(false);
+    }
+  };
+
+  const handleFlagged = () => {
+    setFlagged(!flagged);
+    setMoveOutDate(dateLeft? new Date(Date.parse(dateLeft)) : undefined);
+  };
+
+  const handleRoomNumberChange = (e: { target: { value: unknown } }) => {
+    const inputValue = e.target.value as string;
+    if (inputValue !== null && /^[0-9]{0,3}$/.test(inputValue)) {
+      setRoomNumber(parseInt(inputValue, 10));
+      setRoomNumberError(false);
+    }
+  };
+
+  const handleMoveInDateChange = (inputValue: Date) => {
+    if (inputValue !== null) {
+      if (moveOutDate && inputValue > moveOutDate) {
+        setMoveInDateError(true);
+      }
+      else {
+        setMoveInDate(inputValue);
+        setMoveInDateError(false);
+      }
+    }
+  };
+  
+  const handleMoveOutDateChange = (inputValue: Date) => {
+    if (inputValue !== null) {
+      if (inputValue >= moveInDate) {
+        setMoveOutDate(inputValue);
+        setMoveOutDateError(false);
+      }
+      else {
+        setMoveOutDateError(true);
+      }
+    } 
+  };
+
+  const handleBuildingChange = (
+    selectedOption: SingleValue<{ label: string; value: string }>,
+  ) => {
+    if (selectedOption !== null) {
+      setUserBuilding(selectedOption.value);
+      setBuildingError(false);
+    }
+  };
+
+  const handleSubmit = () => {
+    setInitialsError(initials.length !== 2);
+    setRoomNumberError(roomNumber.toString().length !== 3);
+    setBuildingError(userBuilding === "");
+
+    //  Prevents form submission if any required values are incorrect
+    if (
+      initials.length !== 2 ||
+      moveInDateError || 
+      roomNumberError ||
+      userBuilding === "" ||
+      moveOutDateError
+    ) {
+      return;
+    }
+    editRes();
+    setShowAlert(true);
```

Comment:
We need the following to ensure the modal actually closes once we hit save
```suggestion
    setShowAlert(true);
    toggleClose();
```